game-core/src/main/java/games/strategy/triplea/delegate/TechTracker.java

```diff
@@ -59,8 +56,7 @@ private TechTracker() {}
     return technologyFrontiers;
   }
 
-  public static synchronized void addAdvance(final PlayerID player, final IDelegateBridge bridge,
-      final TechAdvance advance) {
+  public static void addAdvance(final PlayerID player, final IDelegateBridge bridge, final TechAdvance advance) {
```

Comment:
Is there any chance this method is being invoked concurrently on the same object? 
My concern is that removing the synchronized keyword might cause concurrency issues.

Reply:
That is indeed a possibility.  I scanned several call levels deep, and all methods invoked appear to be thread safe (they are either immutable or some lower level call acquires an appropriate lock).

So, I can't guarantee 100% that there won't be a concurrency issue, but synchronization via the `TechTracker` class lock would be wrong at best. :smile:  If anything, it's just hiding some improper synchronization that we should fix, and could even possibly lead to a deadlock due to incorrect lock ordering.

Reply:
90% Guarantee is enough for me ^^

Reply:
I'll give you 90.1% just to tip the balance. :smile: